llvm-external-projects/iree-dialects/lib/Transforms/TransformMatchers.cpp

```diff
@@ -1149,6 +1181,79 @@ transform_ext::StructuredOpMatcher::passThroughOp() {
   });
 }
 
+transform_ext::StructuredOpMatcher &
+transform_ext::StructuredOpMatcher::hasContractionBody(
```

Comment:
There is a somewhat generic `matchReduction` util in SliceAnalysis that we already use for Linalg vectorization and other places, can we use that ?

Reply:
Contraction has an additional operation that the reduction doesn't, so no, we cannot use that one.